core/lib/Cardano/Address/Style/Shelley.hs

```diff
@@ -507,6 +507,23 @@ inspectShelleyAddress addr
                -- 1000: byron address
                 0b10000000 ->
                     inspectByronAddress addr <|> inspectIcarusAddress addr
+               -- 1110: reward account: keyhash28
+                0b11100000 | BS.length rest == pubkeyHashSize->
+                    Just $ Json.object
+                        [ "address_style"     .= Json.String "Shelley"
+                        , "stake_reference"   .= Json.String "by value"
+                        , "stake_key_hash"    .= base16 (BS.take pubkeyHashSize rest)
+                        , "network_tag"       .= network
+                        ]
+               -- 1111: reward account: scripthash28
+                0b11110000 | BS.length rest == scriptHashSize ->
+                    Just $ Json.object
+                        [ "address_style"     .= Json.String "Shelley"
+                        , "stake_reference"   .= Json.String "by value"
+                        , "stake_key_hash"    .= base16 (BS.take scriptHashSize rest)
+                        , "script_hash"       .= base16 (BS.take scriptHashSize rest)
```

Comment:
In the case of stake address, there is either a script hash or a public stake key hash, there can't be both.

Reply:
I'm still struggling to understand what those all are, even after reading the design spec :hot_face: